lua/nvim-treesitter/refactor/smart_rename.lua

```diff
@@ -0,0 +1,70 @@
+-- Binds a keybinding to smart rename definitions and usages.
+-- Can be used directly using the `smart_rename` function.
+
+local ts_utils = require'nvim-treesitter.ts_utils'
+local configs = require'nvim-treesitter.configs'
+local api = vim.api
+
+local M = {}
+
+function M.smart_rename(bufnr)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  local node_at_point = ts_utils.get_node_at_cursor()
+
+  if not node_at_point then
+    vim.fn.echoerr('No node to rename!')
```

Comment:
i've got unknown function, when was this added to neovim ?

Reply:
You're right. It's a command not a function...

Reply:
Actually, I think it might be to much for a simple warning. I'll explore using a specific highlight instead for the message.

Reply:
I created a method to print a warning msg. `utils.print_warning`. Took it from the vim docs for `echohl`